test/SlowTests/Voron/Issues/RavenDB_16536.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using FastTests.Voron;
+using Sparrow;
+using Voron;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Voron.Issues
+{
+    public class RavenDB_16536 : StorageTest
+    {
+        public RavenDB_16536(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        protected override void Configure(StorageEnvironmentOptions options)
+        {
+            options.ManualFlushing = true;
+            options.ManualSyncing = true;
+            options.MaxNumberOfPagesInJournalBeforeFlush = 2; // low value to ensure it will flush
+        }
+
+        [Fact]
+        public void ShouldNotSayThatThereIsNothingToFlush()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                using (var tx = Env.WriteTransaction())
+                {
+                    var tree = tx.CreateTree("items");
+
+                    tree.Add("items/" + i, new byte[] { 1, 2, 3 });
+
+                    tx.Commit();
+                }
+            }
+
+            var onLogsAppliedMre = new ManualResetEventSlim(false);
+
+            Env.OnLogsApplied += () => onLogsAppliedMre.Set();
+
+            var ownFlusher = new GlobalFlushingBehavior();
```

Comment:
why do you create an instance of this?

Reply:
Because I want to manage and control it explicitly in the test